aiida/orm/groups.py

```diff
@@ -269,7 +273,7 @@ def nodes(self):
         the respective AiiDA subclasses of Node, and also allows to ask for
         the number of nodes in the group using len().
         """
-        return self._backend_entity.nodes
+        return map(convert.get_orm_entity, self._backend_entity.nodes)
```

Comment:
We should use a custom Iterator class as it is done in the Backend classes to have __len__ pipe through the .count() method of the ORM, for efficiency

Reply:
Done